### PR TITLE
[MAINT] improve workflow to test on nightly dependencies

### LINKED ISSUE #4358: Set up "nightly build test" workflow
I would suggest reverting the systematic numpy2 checks on every PR because this is adding noise to our review process (in most cases the failures are unrelated to the PR).

This workflow was run on every PR because not all members of the dev team were notified in case of failures.

I would suggest something I have seen in another repo (can't find it at the moment):
- rename the "numpy2" workflow to  "nightly-build dependencies" because it actually tries to run on after installing all the dependencies in their 'bleeding edge' version 
- run the workflow at regular interval
- if it fails open an issue (unless one is already open) and ping the dev team

_Originally posted by @Remi-Gau in https://github.com/nilearn/nilearn/issues/4309#issuecomment-2031302259_

### PR DESCRIPTION
**Pull request by Remi-Gau**
- Closes #4358 

Changes proposed in this pull request:

Add an action to run test suite using the nightly release of Nilearn dependencies. 1Runs on main (by a push or on manual trigger from the `Action` tab) or from a PR if commit message includes `[test nightly]`.

When running on main, if the workflow fails the action will open an issue using an issue template.

Example of issue: https://github.com/Remi-Gau/nilearn/issues/59

>[!Note]
>
> This PR will only start opening issues if this workflow fails after it has been merged.
>

### CODE REVIEW

File: .github/workflows/README.md [hunk 1]
```diff
@@ -94,7 +94,9 @@ With a merge on main, both "host" and "deploy" jobs are run.
 
 ### check_gha_workflow.yml
 
-Pings github API to collect information about how long each run of the test suite lasted.
+Pings github API to collect information about:
+- how long each run of the test suite lasted,
+- how long the build of the doc lasted.
```
Comment (Remi-Gau) [inline .github/workflows/README.md:99]: Had forgotten to update this.

**Review by bthirion — APPROVED** [review]
LGTM, in the sense  that I cannot spot major issue. But clearly, for this kind of PR you need to merge to know whether it's OK...

**Comment by Remi-Gau** [comment]
> LGTM, in the sense that I cannot spot major issue. But clearly, for this kind of PR you need to merge to know whether it's OK...

yup I expect we may have to tweak a few things when we see how it behaves "in production"

**Comment by Remi-Gau** [comment]
Will merge this one to be able to test #4394 properly

**Pull request merged**